spec.emu

```diff
@@ -20,54 +20,123 @@ contributors: Ujjwal Sharma, Younies Mahmoud
+          <td>[[MinutesDisplay]]</td>
           <td>`"minutes"`</td>
         </tr>
         <tr>
           <td>[[Seconds]]</td>
+          <td>[[SecondsStyle]]</td>
+          <td>[[SecondsDisplay]]</td>
           <td>`"seconds"`</td>
         </tr>
         <tr>
           <td>[[Milliseconds]]</td>
+          <td>[[MillisecondsStyle]]</td>
+          <td>[[MillisecondsDisplay]]</td>
           <td>`"milliseconds"`</td>
         </tr>
         <tr>
           <td>[[Microseconds]]</td>
+          <td>[[MicrosecondsStyle]]</td>
+          <td>[[MicrosecondsDisplay]]</td>
           <td>`"microseconds"`</td>
         </tr>
         <tr>
           <td>[[Nanoseconds]]</td>
+          <td>[[NanosecondsStyle]]</td>
+          <td>[[NanosecondsDisplay]]</td>
           <td>`"nanoseconds"`</td>
         </tr>
       </table>
     </emu-table>
 
-    <emu-note type=editor>The [[Weeks]] field is still under discussion in Temporal. For more, check out the <a href="https://github.com/tc39/proposal-temporal/issues/532">Github thread</a> on the Temporal issue tracker.</emu-note>
+    <emu-clause id="sec-getunitoptions" aoid="GetUnitOptions">
+      <h1>GetUnitOptions ( _unit_, _options_, _baseStyle_, _prevStyle_ )</h1>
+      <p>The GetUnitOptions abstract operation is called with the arguments _unit_ (which must be a String), _options_ (which must be an Object), _baseStyle_ (which must be a String), and _prevStyle_ (which must be a String) and returns a Record containing the relevant options for that unit. The following steps are taken:</p>
+
+      <emu-alg>
+        1. Let _stylesList_ be the List &laquo; `"long"`, `"short"`, `"narrow"` &raquo;.
+        1. If _unit_ is one of `"hours"`, `"minutes"`, `"seconds"`, `"milliseconds"`, `"microseconds"`, or `"nanoseconds"`, then
+          1. Set _stylesList_ to the list-concatenation of _stylesList_ and &laquo; "numeric" &raquo;.
+          1. If _unit_ is one of `"hours"`, `"minutes"`, or `"seconds"`, then
+            1. Set _stylesList_ to the list-concatenation of _stylesList_ and &laquo; "2-digit" &raquo;.
+        1. Let _style_ be ? GetOption(_options_, _unit_, _stylesList_, *undefined*).
+        1. If _style_ is *undefined*, then
+          1. If _baseStyle_ is `"digital"`, then
+            1. If _unit_ is one of `"hours"`, `"minutes"`, `"seconds"`, `"milliseconds"`, `"microseconds"`, or `"nanoseconds"`, then
+              1. Set _style_ to `"numeric"`.
+            1. Else,
+              1. Set _style_ to `"narrow"`.
+          1. Else,
+            1. Set _style_ to _baseStyle_.
+          1. Let _displayDefault_ to `"auto"`.
+        1. Else,
+          1. Let _displayDefault_ to `"always"`.
+        1. Let _displayField_ be the string-concatenation of _unit_ and `"Display"`.
+        1. Let _display_ be ? GetOption(_options_, _displayField_, &laquo; `"auto"`, `"always"` &raquo;, _displayDefault_).
+        1. If _prevStyle_ is `"numeric` or `"2-digit`, then
```

Comment:
Are `numeric` and `2-digit` valid styles for days and larger units?  If not, then should there be logic here in this AO to exclude them?